crates/iceberg/src/scan.rs

```diff
@@ -981,13 +981,14 @@ mod tests {
     use crate::table::Table;
     use crate::TableIdent;
 
-    struct TableTestFixture {
+    pub struct TableTestFixture {
         table_location: String,
-        table: Table,
+        pub table: Table,
```

Comment:
Make this `pub` so that we can share it in the `metadata_scan` test.
We can move it to other places like `test_utils`, but I want to avoid unnecessary changes in the PR.